### PR TITLE
[BOLT] Add support for Linux kernel static keys jump table

### PR DESCRIPTION
**Pull request by maksfb**
Runtime code modification used by static keys is the most ubiquitous
self-modifying feature of the Linux kernel. The idea is to to eliminate
the condition check and associated conditional jump on a hot path if
that condition (based on a boolean value of a static key) does not
change often. Whenever they condition changes, the kernel runtime
modifies all code paths associated with that key flipping the code
between nop and (unconditional) jump.

### CODE REVIEW

**Review by aaupov — APPROVED** [review]
Great work.

File: bolt/include/bolt/Core/MCPlusBuilder.h [hunk 1]
```diff
@@ -1208,6 +1218,14 @@ class MCPlusBuilder {
     return nullptr;
   }
 
+  /// Return MCSymbol that represents a target of this instruction at a given
+  /// operand number \p OpNum. If there's no symbol associated with
+  /// the operand - return nullptr.
+  virtual MCSymbol *getTargetSymbol(MCInst &Inst, unsigned OpNum = 0) const {
```
Comment (ayermolo) [inline bolt/include/bolt/Core/MCPlusBuilder.h:1224]: Is this needed?
Looks like it's only used in. Can that be const?

```
OS << "\tjit\t" << MIB->getTargetSymbol(Instruction)->getName()
       << " # ID: " << DynamicID;
```

**Review by ayermolo — APPROVED** [review]

File: bolt/include/bolt/Core/MCPlusBuilder.h (hunk 1 above)
Comment (maksfb) [reply bolt/include/bolt/Core/MCPlusBuilder.h]: Not needed anymore.

**Review by dcci — COMMENTED** [review]
It's great to see this coming along. I don't have anything to add.

**Pull request merged**